AppController/djinn.rb

```diff
@@ -1541,100 +1538,39 @@ def stop_app(app_name, secret)
     FileUtils.rm_rf("#{PERSISTENT_MOUNT_POINT}/apps/#{app_name}.tar.gz")
     CronHelper.clear_app_crontab(app_name)
 
-    # app shutdown process can take more than 30 seconds
-    # so run it in a new thread to avoid 'execution expired'
-    # error messages and have the tools poll it
+    # Since stopping an application can take some time, we do it in a
+    # thread.
     Thread.new {
-      # The login node has extra stuff to do: remove the xmpp listener and
-      # inform the other nodes to stop the application, and remove the
-      # application from the metadata (soap_server).
-      if my_node.is_login?
-        begin
-          uac = UserAppClient.new(my_node.private_ip, @@secret)
-          if not uac.does_app_exist?(app_name)
-            Djinn.log_info("(stop_app) #{app_name} does not exist.")
-          else
-            result = uac.delete_app(app_name)
-            Djinn.log_debug("(stop_app) delete_app returned: #{result}.")
-          end
-        rescue FailedNodeException
-          Djinn.log_warn("(stop_app) delete_app: failed to talk " +
-            "to the UserAppServer.")
-        end
-        @nodes.each { |node|
-          next if node.private_ip == my_node.private_ip
-          if node.is_appengine?
-            ip = node.private_ip
-            acc = AppControllerClient.new(ip, @@secret)
-            begin
-              result = acc.stop_app(app_name)
-              Djinn.log_debug("Removing application #{app_name} from #{ip} " +
-                "returned #{result}")
-            rescue FailedNodeException
-              Djinn.log_warn("Could not remove application #{app_name} from " +
-                "#{ip} - moving on to other nodes.")
-            end
-          end
-        }
-        pid_files = HelperFunctions.shell("ls #{APPSCALE_CONFIG_DIR}/xmpp-#{app_name}.pid").split
-        unless pid_files.nil? # not an error here - XMPP is optional
-          pid_files.each { |pid_file|
-            pid = HelperFunctions.read_file(pid_file)
-            Djinn.log_run("kill -9 #{pid}")
-          }
+      begin
+        uac = UserAppClient.new(my_node.private_ip, @@secret)
+        if not uac.does_app_exist?(app_name)
```

Comment:
In this case I thought that we may want to disable the app, even if it doesn't have any appserver running. IN case it is already disabled it would be idempotent.